### PR TITLE
SAK-49133 Portal collapsing or expanding sidebar generates a stacktrace

### PR DESCRIPTION
**Pull request by kunaljaykam**
https://sakaiproject.atlassian.net/browse/SAK-49133

### CODE REVIEW

File: library/src/webapp/js/portal/portal.sidebar.collapse.js [hunk 1]
```diff
@@ -29,11 +29,15 @@ class SidebarCollapseButton {
   }
 
   async setCollapsed(collapsed) {
+    if (!portal.user || !portal.user.id) {
```
Comment (adrianfish) [inline library/src/webapp/js/portal/portal.sidebar.collapse.js:32]: You should be able to use !portal?.user?.id here.

File: library/src/webapp/js/portal/portal.sidebar.collapse.js [hunk 2]
```diff
@@ -29,11 +29,15 @@ class SidebarCollapseButton {
...
+        return; // Exit the function early if the user is not logged in.
```
Comment (adrianfish) [inline library/src/webapp/js/portal/portal.sidebar.collapse.js:33]: try and stick to two spaces in new javascript. 2 spaces is a lot better when you write functional code. functional code has a lot of nesting, potentially.

File: library/src/webapp/js/portal/portal.sidebar.collapse.js [hunk 3]
```diff
@@ -29,11 +29,15 @@ class SidebarCollapseButton {
...
+    }
 
     collapsed = collapsed ? "true" : "false";
     const putReq = await fetch(`/direct/userPrefs/updateKey/${portal.user.id}/sakai:portal:sitenav?sidebarCollapsed=${collapsed}`, { method: "PUT" });
     if (!putReq.ok) {
-      console.error(`Could not set collapsed state "${collapsed}" for sidebar.`);
+        console.error(`Could not set collapsed state "${collapsed}" for sidebar.`);
```
Comment (adrianfish) [inline library/src/webapp/js/portal/portal.sidebar.collapse.js:39]:
```suggestion
      console.error(`Could not set collapsed state "${collapsed}" for sidebar.`);
```

**Review by adrianfish — CHANGES_REQUESTED** [review]
Just some small stuff.

File: library/src/webapp/js/portal/portal.sidebar.collapse.js (hunk 1 above)
Comment (kunaljaykam) [reply library/src/webapp/js/portal/portal.sidebar.collapse.js]: Good one!

File: library/src/webapp/js/portal/portal.sidebar.collapse.js [hunk 4]
```diff
@@ -29,11 +29,15 @@ class SidebarCollapseButton {
...
+      return; // Exit the function early if the user is not logged in.
+    }
 
     collapsed = collapsed ? "true" : "false";
     const putReq = await fetch(`/direct/userPrefs/updateKey/${portal.user.id}/sakai:portal:sitenav?sidebarCollapsed=${collapsed}`, { method: "PUT" });
     if (!putReq.ok) {
       console.error(`Could not set collapsed state "${collapsed}" for sidebar.`);
     }
   }
+
```
Comment (adrianfish) [inline library/src/webapp/js/portal/portal.sidebar.collapse.js:42]:
```suggestion
```

**Review by adrianfish — APPROVED** [review]

**Pull request merged**